lib/src/facts/openbsd/processor_resolver.cc

```diff
@@ -0,0 +1,67 @@
+#include <internal/facts/openbsd/processor_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    processor_resolver::data processor_resolver::collect_data(collection& facts)
+    {
+        auto result = posix::processor_resolver::collect_data(facts);
+        size_t len;
+        int mib[2];
+        mib[0] = CTL_HW;
+
+        // Get the physical count of processors
+        len = sizeof(result.logical_count);
+        mib[1] = HW_NCPUFOUND;
```

Comment:
The comment says `physical` but the variable is called `logical_count` but then the debug message says `physical`.

So I went to http://www.openbsd.org/cgi-bin/man.cgi/OpenBSD-current/man3/sysctl.3?query=sysctl&sec=3 to see how `HW_NCPUFOUND` and `HW_NCPU` are defined:

```
HW_NCPU
The number of CPUs being used.
HW_NCPUFOUND
The number of CPUs found.
```

And from that I'm not clear at all that used/found maps to physical/logical or v-v.

And then I found this thread: http://comments.gmane.org/gmane.os.openbsd.misc/181238, where Theo de Raadt's response to a question about determining the number of cores ("logical" processors in our nomenclature) is "You look at dmesg, and then at your processor manual."

So ... I'm confused as to whether we can retrieve logical processor count via sysctl, and if so, how.